tools/toolchains/gcc.py

```diff
@@ -80,8 +84,21 @@ def __init__(self, target,  notify=None, macros=None,
             self.cpu.append("-mfloat-abi=hard")
             self.cpu.append("-mno-unaligned-access")
 
+        # Coverage is turned On if coverage_filter is not empty. This means all sources are compiled with coverage
+        # macro and application is linked with coverage flags.
+        # Only source files that match regex from coverage_filter list are compiled with coverage flags. Since turning
+        # On code coverage on all sources increases static data size it overlaps with stack allocation and causes linker
+        # error.
+        self.coverage_filter = coverage_filter
+        if self.coverage_filter:
+            self.macros.append(self.COVERAGE_MACRO)
```

Comment:
Tiny little Nit: Maybe just put the `self.macros.append(self.COVERAGE_MACRO)` the `if self.coverage_filter:` block below? Is the order important here? Just seems like one more `if` statement.

Not that important, but if you're making changes already just thought I'd suggest this one.